File: 2026POTD/JANUARY/17-01-2026/3047.cs
public class Solution {
    public long LargestSquareArea(int[][] bottomLeft, int[][] topRight) {
        int n = bottomLeft.Length;
        long ans = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int x1 = Math.Max(bottomLeft[i][0], bottomLeft[j][0]);
                int y1 = Math.Max(bottomLeft[i][1], bottomLeft[j][1]);
                int x2 = Math.Min(topRight[i][0], topRight[j][0]);
                int y2 = Math.Min(topRight[i][1], topRight[j][1]);
                if (x2 > x1 && y2 > y1) {
                    long side = Math.Min(x2 - x1, y2 - y1);
                    ans = Math.Max(ans, side * side);
                }
            }
        }
        return ans;
    }
}

File: 2026POTD/JANUARY/17-01-2026/3047.rs
impl Solution {
    pub fn largest_square_area(bottom_left: Vec<Vec<i32>>, top_right: Vec<Vec<i32>>) -> i64 {
        let n = bottom_left.len();
        let mut ans: i64 = 0;
        for i in 0..n {
            for j in (i + 1)..n {
                let x1 = bottom_left[i][0].max(bottom_left[j][0]);
                let y1 = bottom_left[i][1].max(bottom_left[j][1]);
                let x2 = top_right[i][0].min(top_right[j][0]);
                let y2 = top_right[i][1].min(top_right[j][1]);
                if x2 > x1 && y2 > y1 {
                    let side = (x2 - x1).min(y2 - y1) as i64;
                    let area = side * side;
                    if area > ans {
                        ans = area;
                    }
                }
            }
        }
        ans
    }
}